lib/jekyll/readers/post_reader.rb

```diff
@@ -34,19 +34,9 @@ def read_posts(dir)
     #
     # Returns nothing.
     def read_publishable(dir, magic_dir, matcher)
-      read_content(dir, magic_dir, matcher).tap { |docs| docs.each(&:read) }
-        .select do |doc|
-          if doc.content.valid_encoding?
-            site.publisher.publish?(doc).tap do |will_publish|
-              if !will_publish && site.publisher.hidden_in_the_future?(doc)
-                Jekyll.logger.debug "Skipping:", "#{doc.relative_path} has a future date"
-              end
-            end
-          else
-            Jekyll.logger.debug "Skipping:", "#{doc.relative_path} is not valid UTF-8"
-            false
-          end
-        end
+      read_content(dir, magic_dir, matcher)
+        .tap { |docs| docs.each(&:read) }
+        .select(&method(:processable?))
```

Comment:
@lxxxvi I know that the use of `&method` is a clever shortcut. But can we have the "conventional" block form instead..?
(unless you can produce some benchmarks that show using `&method` is a faster alternative..)
```ruby
      read_content(dir, magic_dir, matcher)
        .tap { |docs| docs.each(&:read) }
        .select { |doc| processable?(doc) }
```